src/main/java/org/lantern/http/JettyLauncher.java

```diff
@@ -165,9 +168,11 @@ protected void doGet(HttpServletRequest req,
                 } else if (uri.startsWith("/proxy_google")) {
                     writeFileToResponse(resp, Proxifier.PROXY_GOOGLE);
                 } else {
-                    resp.addCookie(new Cookie("XSRF-TOKEN", model.getXsrfToken()));
-                    if (!resp.containsHeader("Content-Security-Policy")) {
-                        LanternUtils.addCSPHeader(resp);
+                    if (!LanternUtils.isDevMode() && !LanternUtils.isLanternPi()) {
+                        resp.addCookie(new Cookie("XSRF-TOKEN", model.getXsrfToken()));
+                        if (!resp.containsHeader("Content-Security-Policy")) {
+                            LanternUtils.addCSPHeader(resp);
+                        }
```

Comment:
That would probably work, although we'd have to include both 127.0.0.1 and the local network address in order to keep all the current stuff working. With any caller already having to somehow guess the random path and port to even get here, however, to me the whole inclusion of CSP is pretty dubious. We've also seen it cause bugs even on localhost, and it's at least the first thing that fails on IE.